`make` does not work at master (4aea08196fb9d3dcf365192f4bf7da7d3813a23d)

## Environment (please complete the following information):
 - OS: macOS
 - (Neo)Vim version: n/a
 - vim-clap version: master (4aea08196fb9d3dcf365192f4bf7da7d3813a23d)
 - Have you reproduced with a minimal vimrc: n/a
 - Have you updated to the latest plugin version: yes
 - Have you upgraded to/compiled the latest Rust binary: that's what this issues is about

## Describe the bug

`cargo build --release` fails.

## To Reproduce

```console
> make # or cargo build --release
error: failed to determine package fingerprint for build script for icon v0.1.0 (/Users/austin/.local/share/vim/site/pack/packager/start/vim-clap/crates/icon)

Caused by:
  failed to determine the most recently modified file in /Users/austin/.local/share/vim/site/pack/packager/start/vim-clap/crates/icon

Caused by:
  failed to determine list of files in /Users/austin/.local/share/vim/site/pack/packager/start/vim-clap/crates/icon

Caused by:
  failed to open git index at /Users/austin/.local/share/vim/site/pack/packager/start/vim-clap/.git/

Caused by:
  invalid data in index - calculated checksum does not match expected; class=Index (10)
make: *** [maple] Error 101
```

## Expected behavior

`maple` builds correctly.

I have a workaround by fixing #946 to download the pre-built binary, but it would be nice to be able to build this manually.

> Can't reproduce it on my machine. This error info `calculated checksum does not match expected` looks suspicious, have you tried purging the repo and downloading it again?

I just removed vim-clap from its location and told `kristijanhusak/vim-packager` to install uninstalled packages. It cloned the repo fresh and ran `:Clap install-binary!` with this as the output (this also happens with `:Clap install-binary!` *not* run by vim-packager).

```
error: failed to determine package fingerprint for build script for icon v0.1.0 (/Users/austin/.local/share/vim/site/p
ack/packager/start/vim-clap/crates/icon)

Caused by:
  failed to determine the most recently modified file in /Users/austin/.local/share/vim/site/pack/packager/start/vim-c
lap/crates/icon

Caused by:
  failed to determine list of files in /Users/austin/.local/share/vim/site/pack/packager/start/vim-clap/crates/icon

Caused by:
  failed to open git index at /Users/austin/.local/share/vim/site/pack/packager/start/vim-clap/.git/

Caused by:
  invalid data in index - calculated checksum does not match expected; class=Index (10)
```

I don’t do much with rust development, but I do `rustup` periodically and I’m on 1.68.0. Updating to 1.68.2 and running `make` in `vim-clap` results in the same thing:

```console
❯ make
cargo build --release
error: failed to determine package fingerprint for build script for icon v0.1.0 (/Users/austin/.local/share/vim/site/pack/packager/start/vim-clap/crates/icon)

Caused by:
  failed to determine the most recently modified file in /Users/austin/.local/share/vim/site/pack/packager/start/vim-clap/crates/icon

Caused by:
  failed to determine list of files in /Users/austin/.local/share/vim/site/pack/packager/start/vim-clap/crates/icon

Caused by:
  failed to open git index at /Users/austin/.local/share/vim/site/pack/packager/start/vim-clap/.git/

Caused by:
  invalid data in index - calculated checksum does not match expected; class=Index (10)
make: *** [maple] Error 101
```

Adding the `include` line to `crates/icon/Cargo.toml` allows the build to run, but I’m not sure if it’s right (`crates/icon/` has a `build.rs` but does not have a `build = "build.rs"` in `crates/icon/Cargo.toml`).

In case it matters, I’m on macOS 13.3.1 on M1.

```console
> sw_vers
ProductName:		macOS
ProductVersion:		13.3.1
BuildVersion:		22E261
> uname -a
Darwin Austins-MacBook-Pro.local 22.4.0 Darwin Kernel Version 22.4.0: Mon Mar  6 20:59:28 PST 2023; root:xnu-8796.101.5~3/RELEASE_ARM64_T6000 arm64
```

If #946 weren’t also exhibiting, I may have never seen this.

--- a/crates/icon/Cargo.toml
+++ b/crates/icon/Cargo.toml
@@ -5,6 +5,7 @@ authors = ["Liu-Cheng Xu <xuliuchengxlc@gmail.com>"]
 edition = "2021"
 license = "MIT"
 publish = false
+include = ["/Cargo.toml", "src/*.rs"]
 homepage = "https://github.com/liuchengxu/vim-clap"
 description = """
 icon is a tool for drawing an icon according to the path name for vim-clap.

--- a/crates/upgrade/src/lib.rs
+++ b/crates/upgrade/src/lib.rs
@@ -56,7 +56,7 @@ impl Upgrade {
                 }
             }
         } else {
-            println!("No newer release, current maple version: {latest_tag}");
+            println!("No newer prebuilt binary release, current maple version: {local_version}");
         }
 
         Ok(())

--- a/install.sh
+++ b/install.sh
@@ -5,18 +5,6 @@ set -u
 REPO=https://github.com/liuchengxu/vim-clap
 APP=maple
 
-remote_latest_tag() {
-  git -c 'versionsort.suffix=-' ls-remote --exit-code --refs --sort='version:refname' --tags "$REPO" 'v0.*' \
-    | tail --lines=1 \
-    | cut --delimiter='/' --fields=3
-}
-
-local_tag() {
-  "bin/$APP" version | cut --delimiter=' ' --fields=4 | cut --delimiter='-' --fields=1
-}
-
-remote_latest_tag=$(remote_latest_tag)
-
 exists() {
   command -v "$1" >/dev/null 2>&1
 }
@@ -34,7 +22,16 @@ download() {
   fi
 }
 
+remote_latest_tag() {
+  git -c 'versionsort.suffix=-' ls-remote --exit-code --refs --sort='version:refname' --tags "$REPO" 'v0.*' \
+    | tail --lines=1 \
+    | awk -F "/" '{print $NF}'
+}
+
 try_download() {
+  local remote_latest_tag=$(remote_latest_tag)
+  echo "bin/$APP is empty, try downloading the latest prebuilt binary $APP $remote_latest_tag from GitHub ..."
+
   local DOWNLOAD_URL="$REPO/releases/download/$remote_latest_tag"
   local asset=$1
   if [ -z "${TMPDIR+x}" ]; then
@@ -48,7 +45,7 @@ try_download() {
   chmod a+x "bin/$APP"
 }
 
-do_download() {
+download_prebuilt_binary() {
   arch=$(uname -sm)
   case "${arch}" in
       "Linux x86_64")
@@ -60,21 +57,15 @@ do_download() {
       "Darwin arm64")
         try_download "$APP"-aarch64-apple-darwin ;;
       *)
-        echo "No prebuilt maple binary available for ${arch}."
+        echo "No prebuilt maple binary available for this platform ${arch}."
+        echo "You can compile the binary locally by running `make` or `cargo build --release` if Rust has been installed."
         exit 1
         ;;
   esac
 }
 
 if [ ! -f "bin/$APP" ]; then
-  echo "bin/$APP is empty, try downloading $APP $remote_latest_tag from GitHub directly..."
-  do_download
+  download_prebuilt_binary
 else
-  if [ $(local_tag) == remote_latest_tag ]; then
-    echo "Local binary "bin/$APP" is already the latest version"
-    exit 0
-  else
-    echo "Try downloading latest version of $APP $remote_latest_tag from GitHub..."
-    do_download
-  fi
+  "bin/$APP" upgrade
 fi

